fast64_internal/oot/scene/exporter/to_c/scene_table_c.py

```diff
@@ -1,283 +1,257 @@
-
-
-def getHackerOoTCheck(line: str):
-    return (
-        line != "\n"
-        and '#include "config.h"\n' not in line
-        and "#ifdef INCLUDE_TEST_SCENES" not in line
-        and "#endif" not in line
-        and not line.startswith("// ")
-    )
+        if self.selectedSceneEnumValue is not None:
+            self.selectedSceneIndex = self.getIndexFromEnumValue()
 
+        self.entryBySpecName = {entry.specName: entry for entry in self.entries}
 
-def getSceneTable(exportPath):
-    """Read and remove unwanted stuff from ``scene_table.h``"""
-    dataList = []
-    sceneNames = []
-    fileHeader = ""
-
-    # read the scene table
-    try:
-        with open(os.path.join(exportPath, "include/tables/scene_table.h")) as fileData:
-            # keep the relevant data and do some formatting
-            for i, line in enumerate(fileData):
-                # remove empty lines from the file
-                if not line.strip():
-                    continue
-
-                if not bpy.context.scene.fast64.oot.hackerFeaturesEnabled or getHackerOoTCheck(line):
-                    if not (
-                        # Detects the multiline comment at the top of the file:
-                        (line.startswith("/**") or line.startswith(" *"))
-                        # Detects single line comments:
-                        # (meant to detect the built-in single-line comments
-                        #  "// Debug-only scenes" and "// Added scenes")
-                        or line.startswith("//")
-                    ):
-                        dataList.append(line[(line.find("(") + 1) :].rstrip(")\n").replace(" ", "").split(","))
-                    else:
-                        # Only keep comments before the data (as indicated by dataList being empty).
-                        # This prevents duplicating the built-in single-line comments to the header.
-                        # It also means other handwritten single-line comments are removed from the file.
-                        if not dataList:
-                            fileHeader += line
-                if line.startswith("/* 0x"):
-                    startIndex = line.find("SCENE_")
-                    sceneNames.append(line[startIndex : line.find(",", startIndex)])
-    except FileNotFoundError:
-        raise PluginError("ERROR: Can't find scene_table.h!")
-
-    # return the parsed data, the header comment and the comment mentionning debug scenes
-    return dataList, fileHeader, sceneNames
-
-
-def getSceneIndex(sceneNameList, sceneName):
-    """Returns the index (int) of the chosen scene, returns None if ``Custom`` is chosen"""
-    if sceneName == "Custom":
-        return None
-
-    if sceneNameList is not None:
-        for i in range(len(sceneNameList)):
-            if sceneNameList[i] == sceneName:
+    def getIndexFromEnumValue(self):
+        """Returns the index (int) of the chosen scene, returns -1 if custom and -2 if insertion"""
```

Comment:
not a fan, please return some object instead, maybe
```py
class SomeSceneIndexType(enum.Enum):
    VANILLA = enum.auto()
    CUSTOM = enum.auto()
    INSERTION = enum.auto()

@dataclass
class SomeSceneIndex:
    type: SomeSceneIndexType
    index: Optional[int]  # only if type == SomeSceneIndexType.VANILLA
```

(I don't quite understand what "-2" is)

Reply:
I used negative numbers since positive ones may be confused with actual indices